neo/app/assets/javascripts/timeseries/controllers.js

```diff
@@ -0,0 +1,94 @@
+angular.module('cloudberry.timeseries', ['cloudberry.common'])
+  .controller('TimeSeriesCtrl', function ($scope, $window, Asterix) {
+    $scope.result = [];
+    $scope.d3 = $window.d3;
+    $scope.dc = $window.dc;
+    $scope.crossfilter = $window.crossfilter;
+    $scope.preProcess = function (result) {
+      var parseDate = d3.time.format("%Y-%m").parse;
+      var result_array = [];
+      angular.forEach(result, function (value, key) {
+        key = parseDate(key);
+        value = +value;
+        result_array.push({'time':key, 'count':value});
+      });
+      return result_array;
+    };
+    $scope.$watch(
+      function() {
+        return Asterix.timeResult;
+      },
+
+      function(newResult) {
+        if(newResult && Asterix.queryType != 'time') {
+          $scope.result = newResult;
+          $scope.result = $scope.preProcess($scope.result);
+        }
+      }
+    );
+  })
+  .directive('timeSeries', function (Asterix) {
+    var margin = {
+      top: 10,
+      right: 10,
+      bottom: 30,
+      left: 50
+    };
+    var width = 962 - margin.left - margin.right;
```

Comment:
could you replace these constant value `962` and `150` with some relative values?
I know it will look beautiful in the normal screen. However, we need to show the screen on the lower resolution projector, which should also look good.